### DISCUSSION TITLE
Feature Request: Allow for a "wake-up" key press in send_start_channel

### DISCUSSION THREAD

**Discussion by jsn081** [GerryDazoo/Slinger#361] [Ideas]
My use case: I'm using Slingbox on a TiVo (specifically a TiVo Mini).  The mini will fall asleep and go to a screen saver after a while.  I'm attempting to connect Slinger using a 500 to Plex via xTeVe via the instructions listed elsewhere.

My issue is if I send a start channel when the Mini is asleep, it will wake up and assume the number entered from IR is a shortcut to a menu page.  So the channel change doesn't happen, and now I'm stuck in a menu.

If I send a LIVE IR command (remote code 32) and wait two seconds (not totally necessary but helps the Mini to not miss IR commands) everything works as expected.  The benefit in my case is if I resend LIVE when it's already awake and in live TV mode, nothing happens.

I thought about using the `?channel=` feature to send code 32 before the channel (using `+`), but I couldn't think of a way to make Plex do this once, not on every channel change, and not while I'm manually watching via VLC and `/remote`.

My hack fix for this was to go to line [800](https://github.com/GerryDazoo/Slinger/blob/df59690bab465c049c028060a872e0413620482c/slingbox_server.py#L800) in the send_start_channel definition and under the print line insert:

`SendKeycode(str(32), rccode) # send LIVE key`
`time.sleep( 2 ) # wait 2 seconds`

...and it works as expected.  Obviously this is a hack for only my use-case, but it may be worth putting something in the config file to allow for a wake-up key to be allowed and specified.  I can think of some other STBs that would benefit from it.

Now for sanity I'm about to add a check to make sure it only does this once every 30m.

Just my thoughts.   Hopefully this will help someone else trying to do the same thing I'm doing, and it may be worth integrating into the rest of the project.

**Comment by rbf121** [comment]
Someone else asked about this, I forget where the question was asked or if they solved it.  You could look at the keepalive stuff in Gerry's code, lines 965-990 and add an option to send the IR code on startup and then every X minutes, etc.  Make it optional and configurable in config.ini

**Comment by GerryDazoo** [comment]
Or....
With the HTML remote support you could whip up a little java "on-load" script to send that keycode every x minutes.

**Comment by jsn081** [comment]
In my use case that wouldn't work.  Doing the Plex via xTeVe route it only uses the ?channel= tuning method.  If I had access to the HTML remote I wouldn't need to code it because I'd just hit "live" on the remote and be done - not that hard or annoying.  It's the automated tuning where I have the issue.

**Reply by Chaskel** [reply]
Hi @jsn081 - Many thanks for sharing. I'm in a similar situation with an Arris VIP5602W box (for which I use the Bell Fibe VIP2202 remote), and I press OK (42) to wake it up. I wanted to mention that when I press OK when the cable box is already awake, it will briefly show a pop-up, and it disappears a few seconds later (I am not sure if you have similar experience in your setup). I thought I would mention this as I suspect it would be bothersome for anyone in a similar situation to keep seeing a pop-up while watching a channel (if there was a periodic keycode sent to keep it awake).

As I am currently OK with having the command sent for each channel change, I've gone ahead and inserted the following code just before the "digits2buttons = ['18','9','10','11','12','13','14','15','16','17']" line and it seems to work:

SendKeycode(str(42), rccode) # send OK/SELECT key
time.sleep( 2 ) # wait 2 seconds

I'll also add that I believe my previous DISH Network VIP722K satellite receiver required pressing the "SELECT" button after going to sleep.

Thanks again for sharing.
